br/cmd/br/cmd.go

```diff
@@ -107,6 +110,23 @@ func AddFlags(cmd *cobra.Command) {
 	_ = cmd.PersistentFlags().MarkHidden(FlagRedactLog)
 }
 
+func calculateMemoryLimit(memleft uint64) uint64 {
+	const halfGB = uint64(512 * 1024 * 1024)
+	const fourGB = 8 * halfGB
+	// memreserved = f(memleft) = 512MB * memleft / (memleft + 4GB)
```

Comment:
just for my knowledge, what is our memory footprint like in br cli, is max 512MB large enough?

Reply:
Actually, the `memreserved` is not the memory reserved for BR, but it is the memory reserved for system. Make sure the system still has `memreserved` memory to avoid to triggering OOM.
The max memory reserved for BR is `memleft - memreserved`.

For backup, the memory footprint is
```
Domain:
  many tables Info (about 4GB if 1 million tables)

Ranges:
  no more than 1 GB (about 2 million ranges)

Temporary Table Infos BR dealing with:
  no more than 256 MB (kv batch default cache size)

Temporary Schema data to be flushed to external storage:
  no more than 256 MB

(Optional) Statistic:
  no more than schema-concurrency * max memory size of partition/table's statistic

Other Grpc memory: ...
```

For restore, the memory footprint is
```
Domain:
  many tables Info (about 4GB if 1 million tables)

All the backupmeta from external storage:
  many tables info (about 4GB if 1 million tables)
  no more than 1 GB (about 2 million ranges)

(Optional) Statistic: 
  no more than schema-concurrency * max memory size of partition/table's statistic

Goroutines restoring SSTs files: (The workpool is very large)
  some scan region information
  some sst meta
  some grpc memory
  goroutine stack itself
```

Reply:
thanks! That's really useful. 
oh I see, yeah I thought it's the memory reserved for the BR cli.